src/Chess/Game/ChessGame.php

```diff
@@ -1293,15 +1296,16 @@ public function inStaleMate($color = null)
             if (count($canMove)) {
                 $a = $this->getPiece($name);
                 foreach ($canMove as $move) {
-                    $this->startTransaction();
                     $this->_move = $color;
 
                     if (!$this->isError($this->moveSquare($a, $move))) {
+                        $this->rollbackTransaction();
```

Comment:
The entire method is about checking if the game is in stalemate, it should never modify the state of the object. It just needs to know about the internal state at that moment and make an assumption based on that.  By rolling back the transaction in the other cases I'm making sure that the state is never modified.